src/test/java/ipman/ui/SpyUi.java

```diff
@@ -0,0 +1,47 @@
+package ipman.ui;
+
+import ipman.commands.Command;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class SpyUi implements Ui {
+    private final List<String> record = new ArrayList<>();
```

Comment:
Perhaps plural form should be used for collection of items

Reply:
Hmm, I think this field is a bit ambiguous. I was thinking of record as a collective noun i.e. I keep one record of all the messages shown. I have renamed this field to `recordedMessages` instead of `records`.